frontend/public/components/utils/details-item.tsx

```diff
@@ -98,7 +99,9 @@ export const DetailsItem: React.FC<DetailsItemProps> = ({
             <>
               <SplitItem isFilled />
               <SplitItem>
-                <EditButton onClick={onEdit}>{t('public~Edit')}</EditButton>
+                <EditButton data-test={`details-item-value__${label}`} onClick={onEdit}>
```

Comment:
[nit] This test id doesn't seem to really match the EditButton component. Should be something like `${label}-details-item__edit-button`. Also I would recommend just passing label through to `EditButton` as a prop, then defining the data-test attribute on the `Button` within that component. You'll also need to update the type definition for `EditButton` props.  
```suggestion
                <EditButton id={label} onClick={onEdit}>
```

Reply:
added a `label` prop